File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/CurriculumItemDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for a curriculum item (lecture, quiz, etc.).
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CurriculumItemDTO {
    private Long id;
    private String title;
    private String description;
    private String contentSummary; // e.g., "15:47"
    private String itemType; // "lecture", "quiz", "assignment", "practice_test"
    private Boolean canBePreviewed;
    private Boolean isCodingExercise;
    private Boolean isPracticeTest;
    private String previewUrl;
    private String learnUrl;
    private Integer objectIndex;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/CurriculumItemDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * DTO for a curriculum item (lecture, quiz, etc.).
 */
data class CurriculumItemDTO(
    val id: Long? = null,
    val title: String? = null,
    val description: String? = null,
    // e.g., "15:47"
    val contentSummary: String? = null,
    // "lecture", "quiz", "assignment", "practice_test"
    val itemType: String? = null,
    val canBePreviewed: Boolean? = null,
    val isCodingExercise: Boolean? = null,
    val isPracticeTest: Boolean? = null,
    val previewUrl: String? = null,
    val learnUrl: String? = null,
    val objectIndex: Int? = null,
)
